Program.cs

```diff
@@ -0,0 +1,293 @@
+                else
+                {
+                    foreach (var f in flights)
+                    {
+                        Console.WriteLine($"ID: {f.FlightId} | {f.DepartureCountry} ({f.DepartureAirport}) -> {f.DestinationCountry} ({f.ArrivalAirport}) | Date: {f.DepartureDate:yyyy-MM-dd}");
+                        Console.WriteLine($"Prices: Economy={f.PriceEconomy}, Business={f.PriceBusiness}, First={f.PriceFirstClass}");
+                        Console.WriteLine(new string('-', 30));
+                    }
+                }
+                break;
+
+            case "2":
+                Console.Write("Enter your full name: ");
+                string passengerName = Console.ReadLine() ?? "";
+
+                Console.Write("Enter your passport number: ");
+                string passportNumber = Console.ReadLine() ?? "";
+
+                // Add passenger if not exists
+                var existingPassengers = passengerService.GetAll();
+                if (!existingPassengers.Exists(p => p.PassportNumber == passportNumber))
+                {
+                    passengerService.AddPassenger(new Passenger { Name = passengerName, PassportNumber = passportNumber });
+                }
+
+                Console.Write("Enter Flight ID to book: ");
+                string flightId = Console.ReadLine() ?? "";
+                var flight = flightService.GetAllFlights().Find(f => f.FlightId == flightId);
+                if (flight == null)
+                {
+                    Console.WriteLine("Flight not found.");
+                    break;
+                }
+
+                Console.WriteLine("Choose class:");
+                Console.WriteLine("1. Economy");
+                Console.WriteLine("2. Business");
+                Console.WriteLine("3. FirstClass");
+                string classChoice = Console.ReadLine() ?? "";
+                BookingClass bookingClass = classChoice switch
+                {
+                    "1" => BookingClass.Economy,
+                    "2" => BookingClass.Business,
+                    "3" => BookingClass.FirstClass,
+                    _ => BookingClass.Economy
+                };
+
+                var newBooking = new Booking
+                {
+                    PassengerName = passengerName,
+                    PassportNumber = passportNumber,
+                    FlightId = flight.FlightId,
+                    BookingClass = bookingClass,
+                    BookingDate = DateTime.Now
+                };
+
+                bookingService.AddBooking(newBooking);
+                Console.WriteLine("Booking completed.");
+                break;
+
+            case "3":
+                Console.Write("Enter your passport number to view bookings: ");
+                string passportSearch = Console.ReadLine() ?? "";
+
+                var bookings = bookingService.GetAll();
+                var myBookings = bookings.FindAll(b => b.PassportNumber == passportSearch);
```

Comment:
we shouldn't filter here, we should have this in the booking service (get a booking by passportnumber), and it should have getall and findall.